### PR TITLE
DM-17428: interpImage: allow transposition before interpolation

### PR DESCRIPTION
**Pull request by PaulPrice**
This allows the interpolation to act over columns instead of rows.

### CODE REVIEW

File: python/lsst/pipe/tasks/interpImage.py [hunk 1]
```diff
@@ -168,7 +174,78 @@ def run(self, image, planeName=None, fwhmPixels=None, defects=None):
...
+            return result
+
+        useImage = maskedImage
+        useDefects = defects
+        if self.config.transpose:
+            useImage = afwImage.makeMaskedImage(transposeImage(maskedImage.image),
+                                                transposeImage(maskedImage.mask),
+                                                transposeImage(maskedImage.variance))
+            useDefects = ipIsr.transposeDefectList(defects)
+        try:
```
Comment (yalsayyad) [inline python/lsst/pipe/tasks/interpImage.py]: Unclear to me why you put this in a `try/finally` block here.  If the code in the `with` block that uses this context throws an exception, yield rethrows it (regardless if the try block is here or not). If that happens, what's the point of putting the pre-interpolation pixels back in?  This isn't restoring some previous state.

File: python/lsst/pipe/tasks/interpImage.py [hunk 2]
```diff
@@ -168,7 +174,78 @@ def run(self, image, planeName=None, fwhmPixels=None, defects=None):
...
+        def transposeImage(image):
+            """Transpose an image"""
+            result = type(image)(transposeBox(image.getBBox()))
+            result.array[:] = image.array.T
+            return result
+
+        useImage = maskedImage
+        useDefects = defects
+        if self.config.transpose:
+            useImage = afwImage.makeMaskedImage(transposeImage(maskedImage.image),
```
[the remark this answers is not in the document]
Comment (PaulPrice) [reply python/lsst/pipe/tasks/interpImage.py]: I think that should be obvious from the code. If you disagree, let me know and I'll write something.

[the remark this answers is not in the document]
Comment (yalsayyad) [reply python/lsst/pipe/tasks/interpImage.py]: 👍 Without the try/finally block, it's obvious.   (The finally block had me thinking that you thought you had destroyed `maskedImage`)

File: python/lsst/pipe/tasks/interpImage.py [hunk 3]
```diff
@@ -168,7 +174,78 @@ def run(self, image, planeName=None, fwhmPixels=None, defects=None):
...
+            transposed.
+        """
+        def transposeBox(box):
+            """Transpose a bounding box"""
+            return afwGeom.Box2I(afwGeom.Point2I(box.getMinY(), box.getMinX()),
+                                 afwGeom.Point2I(box.getMaxY(), box.getMaxX()))
+
+        def transposeImage(image):
+            """Transpose an image"""
+            result = type(image)(transposeBox(image.getBBox()))
```
[the remark this answers is not in the document]
Comment (RobertLuptonTheGood) [reply python/lsst/pipe/tasks/interpImage.py]: In particular `type` doesn't work with proxies, although I'm not sure if they'll reappear

**Pull request merged**